crates/bevy_text/src/font.rs

```diff
@@ -16,9 +16,78 @@ use bevy_reflect::TypePath;
 ///
 /// Bevy currently loads a single font face as a single `Font` asset.
 #[derive(Debug, TypePath, Clone, Asset)]
-pub struct Font {
+pub enum Font {
     /// Content of a font file as bytes
-    pub data: Arc<Vec<u8>>,
+    Data(Arc<Vec<u8>>),
+    /// References a font inserted into the font database by family, weight, stretch, and style.
+    ///
+    /// This can include system fonts, if enabled in [`super::TextPlugin`], or previously loaded fonts via [`Font::Data`].
+    Query {
```

Comment:
`Query` is so ubiquitous in Bevy, so I think it's probably better to disambiguate this a bit, even though it's unlikely to be confusing to anyone:
```suggestion
    DbQuery {
```